### DISCUSSION TITLE
How to create Debian package from Milvus source code?

### DISCUSSION THREAD

**Discussion by liorf95** [milvus-io/milvus#33654] [Q&A and General discussion] [answered]
Is there any guide/doc explaining how to create a Debian package from Milvus source code to be deployed as a standalone?

**Comment by yhmo** [comment]
It might be a difficult thing for users.
I don't know the answer, let me ask the owner.

**Reply by liorf95** [reply]
What do you mean by a difficult thing for users? - So how users do deploy Milvus with the published https://github.com/milvus-io/milvus/releases/download/v2.0.2/milvus_2.0.2-1_amd64.deb for example?

Let me please rephrase my question:
How can I find the process which builds the published https://github.com/milvus-io/milvus/releases/download/v2.0.2/milvus_2.0.2-1_amd64.deb? 
My aim is to do code changes in Milvus source code, build the project again with my changes with make,  and publish a private Debian package installation from it.

**Reply by yhmo** [reply]
Standard steps to compile Milvus locally:

1. Make sure your soft environment meets the requirement: https://github.com/milvus-io/milvus/blob/master/DEVELOPMENT.md

2. Enter the project root path, run this script to set environment paths.
`source ./scripts/setenv.sh`

3. Under the project root path, run this script to install the dependent libraries.
`./scripts/install_deps.sh`

4. Under the project root path, run "make" command to compile milvus.
`make`
Once it successfully compiles, you will see ./bin/milvus generated under the project root path.


Wait the installation owner to tell how to make the deb.

**Reply by liorf95** [reply]
Thanks!

**Reply by liorf95** [reply]
BTW - make rpm does create an rpm package from the build but I did not find an explicit way for creating a deb pkg (without converting the rpm externally afterward).

**Reply by pingliu** [reply, marked as answer]
enter the build/deb dir , run the scripts, bash ./build_deb.sh

**Reply by liorf95** [reply]
Yep, I had found it too.
Thanks.
